strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/AllChanges.java

```diff
@@ -0,0 +1,39 @@
+package org.arquillian.smart.testing.vcs.git;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The intention of this class is not to provide a final strategy to final user but to be used by other strategies that requires all changes of Git commits.
+ */
+public class AllChanges {
+
+    private NewFilesDetector newFilesDetector;
+    private ChangedFilesDetector changedFilesDetector;
+
+    public AllChanges(File currentDir, String previous, String head, String ... globPatterns) {
+        newFilesDetector = new NewFilesDetector(currentDir, previous, head, globPatterns);
+        changedFilesDetector = new ChangedFilesDetector(currentDir, previous, head, globPatterns);
+    }
+
+    public Set<String> getTests() {
+        final Set<String> tests = new HashSet<>();
+
+        tests.addAll(newFilesDetector.getTests());
+        tests.addAll(changedFilesDetector.getTests());
```

Comment:
Yes this was also mine but I notice that if we do this refactor maybe the code becomes less clear, so we can do two things the first one accept it and if we need to do in the future we have the class, or just drop the PR or do a full refactor which I am not totally sure now.